### ISSUE TITLE
false positive for mut-mut when mutably borrowing iterator

### ISSUE THREAD

**Issue by oli-obk** [Manishearth/rust-clippy#939]
The span is on the for loop, so maybe it's occurring inside the expansion?

```rust
    let array = [5, 6, 7, 8, 9];
    let mut args = array.iter().skip(2);
    for &arg in &mut args {
        println!("{}", arg);
        if arg == 8 {
            break;
        }
    }
    for arg in args {
        println!(":{}", arg);
    }
```

**Comment by oli-obk** [comment]
hmm... that should be caught by `in_external_macro`, but it isn't.
